src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -948,7 +955,7 @@ public TFTensor GetTensor()
             {
                 _srcgetter(ref _vBuffer);
 
-                Utils.EnsureSize(ref _denseData, _vBuffer.Length, keepOld: false);
+                _denseData = new T[_vBuffer.Length];
```

Comment:
Are you sure you always want to allocate a new array here? If yes, then why hold on to `_denseData`?

Reply:
I spotted it as a problem when I saw _denseData has four elements instead of 2 when `_vBuffer.Length=2`. It's actually fine if we used vBuffer.Length to access elements in _denseData. I am reverting it back to original with comments detailing the logic.

---
In reply to: [256866967](https://github.com/dotnet/machinelearning/pull/2546#discussion_r256866967) [](ancestors = 256866967)